### PR TITLE
feat: properties

### PR DESCRIPTION
**Pull request by DasProffi**
## Which issues does this pull request close?
#888
#889

## [OPTIONAL] Give testing instructions to reviewers
Look at the story book

## Context
from #862

### CODE REVIEW

**Comment by DasProffi** [comment]
#862 is split and I opened Issues for the missing feature #888, #889, #890

File: lib/components/Properties/PropertyBase.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import type { ReactNode } from 'react'
+import { AlertTriangle } from 'lucide-react'
+import { tw, tx } from '../../twind'
+
+export type PropertyBaseProps = {
+  name: string,
+  input: (props: { required: boolean, hasValue: boolean }) => ReactNode,
+  hasValue: boolean,
+  required?: boolean,
```
Comment (MaxSchaefer) [inline lib/components/Properties/PropertyBase.tsx:9]:
```suggestion
  softRequired?: boolean,
```

File: lib/components/user-input/Textarea.tsx [hunk 2]
```diff
@@ -34,6 +38,9 @@ export const Textarea = ({
   disclaimer,
   onBlur,
   onEditCompleted = noop,
+  defaultStyle = true,
+  className,
+  outerClassName = '',
```
Comment (MaxSchaefer) [inline lib/components/user-input/Textarea.tsx:43]: Not sure about this `outerClassName`. Why does the caller cannot set this around the component?

File: lib/util/storage.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 class StorageSerice {
-  // this seems to be a bug in eslint as 'paramter-properties' is a special syntax of typescript
+  // this seems to be a bug in eslint as 'paramter-Properties' is a special syntax of typescript
```
Comment (MaxSchaefer) [inline lib/util/storage.ts:2]:
```suggestion
  // this seems to be a bug in eslint as 'paramter-properties' is a special syntax of typescript
```

File: lib/components/Properties/DateProperty.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  return (
+    <PropertyBase
+      {...baseProps}
+      hasValue={hasValue}
+      icon={<CalendarDays size={16}/>}
+      input={({ required }) => (
+        <div
+          className={tx('flex flex-row grow py-2 px-4 cursor-pointer', { 'text-hw-warn-600': required && !hasValue })}
+        >
+          <input
```
Comment (MaxSchaefer) [inline lib/components/Properties/DateProperty.tsx:32]: I think this is the second time we need a date input. The first time was in [TaskDetails](https://github.com/helpwave/web/blob/main/tasks/components/layout/TaskDetailView.tsx#L186). Please move this out into a separate component. Can we then also get rid of this calendar icon or make it optional?

File: lib/components/Properties/NumberProperty.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      input={({ required }) => (
+        <div
+          className={tx('flex flex-row grow py-2 px-4 cursor-pointer', { 'text-hw-warn-600': required && !hasValue })}
+        >
+          <Input
+            expanded={false}
+            className={tx('!ring-0 !border-0 !outline-0 !p-0 !m-0 !w-fit !shadow-none !rounded-none', { 'bg-hw-warn-200 placeholder-hw-warn-500': required && !hasValue })}
+            value={value?.toString() ?? ''}
+            type="number"
+            readOnly={!onChange}
```
Comment (MaxSchaefer) [inline lib/components/Properties/NumberProperty.tsx:59]: Should be explicitly set via props

File: lib/components/Properties/DateProperty.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            className={tx('!ring-0 !border-0 !outline-0 !p-0 !m-0', { 'bg-hw-warn-200': required && !hasValue })}
+            value={dateText}
+            type="datetime-local"
+            readOnly={!onChange}
```
Comment (MaxSchaefer) [inline lib/components/Properties/DateProperty.tsx:36]: Should be explicitly set via props

File: lib/components/Properties/TextProperty.tsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+          className={tx('flex flex-row grow pt-2 pb-1 px-4 cursor-pointer', { 'text-hw-warn-600': required && !hasValue })}
+        >
+          <Textarea
+            outerClassName={tw('w-full')}
+            // TODO consider the height and rows for dynamic sizing
+            className={tx('!ring-0 !border-0 !outline-0 !p-0 !m-0 !shadow-none !rounded-none', { 'bg-hw-warn-200 placeholder-hw-warn-500': required && !hasValue })}
+            rows={5}
+            defaultStyle={false}
+            value={value?.toString() ?? ''}
+            readOnly={!onChange}
```
Comment (MaxSchaefer) [inline lib/components/Properties/TextProperty.tsx:58]: Should be explicitly set via props

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]
Please rename `lib/components/Properties` to `lib/components/properties`

File: lib/components/Properties/DateProperty.tsx (hunk 4 above)
Comment (DasProffi) [reply lib/components/Properties/DateProperty.tsx]: WIthout the icon only manual date input is possible and we would have to import or write our own datatime picker

Comment (DasProffi) [reply lib/components/Properties/DateProperty.tsx]: Will be done in #895

**Review by MaxSchaefer — APPROVED** [review]

**Review by MaxSchaefer — APPROVED** [review]

**Pull request merged**